Assets/Scripts/Tutorial/TutorialController.cs

```diff
@@ -42,82 +47,105 @@ private void OnEnable()
-                _currStep = 0;
-                tutorialSteps[_currStep].gameObject.SetActive(true);
-                tutorialSteps[_currStep].Setup(TutorialDashboard.TMP);
-                
-                _hapticsController.Haptics(1, 0.5f, 1, OVRInput.Controller.LTouch);
+                Setup();
             }
             else
             {
-                End();
+                enabled = false;
+            }            
+        }
+        private void Setup()
+        {
+            _notebook.ShowChapter(Chapter.ChapterTitle.Tutorial);                         // display the tutorial chapter
+            _notebook.CurrentChapter.PresentPage(_notebook.CurrentChapter.defaultPage);   // display the default page of the tutorial chapter
+
+            foreach (TutorialStep tutorialStep in tutorialSteps)
+            {
+                tutorialStep.Cleanup();
+                tutorialStep.chapter = _notebook.CurrentChapter;
+                if (tutorialStep.AllowActiveFalse == true) tutorialStep.gameObject.SetActive(false);
             }
+
+            // set up the first tutorial item
+            _currStep = 0;
+            tutorialSteps[_currStep].gameObject.SetActive(true);
+            tutorialSteps[_currStep].Setup();
+            tutorialSteps[_currStep].OnDone.AddListener(Next);
+
+            // update the page count
+            int step = _currStep + 1;
+            _pageCountString = "Tutorial step " + step + " of " + tutorialSteps.Length;
+            _notebook.CurrentChapter.ChangeText(pageCount, _pageCountString);
             
+            _hapticsController.Haptics(1, 0.5f, 1, OVRInput.Controller.LTouch);
         }
 
         public void Next()
         {
             _hapticsController.Haptics(1, 0.5f, 1, OVRInput.Controller.LTouch);
 
             // clean up the current step
-            tutorialSteps[_currStep].Cleanup(TutorialDashboard.TMP);
+            tutorialSteps[_currStep].Cleanup();
             if (tutorialSteps[_currStep].AllowActiveFalse == true) tutorialSteps[_currStep].gameObject.SetActive(false);
 
             // set up the next step
             _currStep += 1;
             if (_currStep < tutorialSteps.Length)
             {
                 tutorialSteps[_currStep].gameObject.SetActive(true);
-                tutorialSteps[_currStep].Setup(TutorialDashboard.TMP);
+                tutorialSteps[_currStep].Setup();
+                tutorialSteps[_currStep].OnDone.AddListener(Next);
+
+                // update the page count
+                int step = _currStep + 1;
+                _pageCountString = "Tutorial step " + step + " of " + tutorialSteps.Length;
+                _notebook.CurrentChapter.ChangeText(pageCount, _pageCountString);
             }
             else
             {
-                End();
+                enabled = false;
             }
         }
 
         public void Skip()
         {
             _skipped = true;
-            End();
+            enabled = false;
```

Comment:
will this really result in OnDisable being called? You're not disabling the game object here, just the script